lib/SSO.php

```diff
@@ -28,17 +29,24 @@ public function getAuthorizationUrl(
         $state,
         $provider,
         $connection,
+        $organization = null,
         $domainHint = null,
         $loginHint = null
     ) {
         $authorizationPath = "sso/authorize";
 
-        if (!isset($domain) && !isset($provider) && !isset($connection)) {
-            $msg = "Either \$domain, \$provider, or \$connection is required";
+        if (!isset($domain) && !isset($provider) && !isset($connection) && !isset($organization)) {
+            $msg = "Either \$domain, \$provider, \$connection, or \$organization is required";
 
             throw new Exception\UnexpectedValueException($msg);
         }
 
+        if (isset($domain) && !isset($provider) && !isset($connection) && !isset($organization)) {
+            $msg = "Domain is being deprecated, please switch to using Connection or Organization ID";
+
+            trigger_error($msg, E_USER_NOTICE);
```

Comment:
Does this throw a true exception or is it just a warning? We'll want it to be just a warning so maintain backwards compatibility.

Reply:
@robframpton  From my research there should be 3 tiers of exception:

E_USER_NOTICE
E_USER_WARNING
E_USER_ERROR

Notice and warning are just messages that won't stop the functionality - https://beamtic.com/e_user_notice-php

Reply:
Here's the official PHP documentation https://www.php.net/manual/en/errorfunc.constants.php

Reply:
Awesome thanks for the docs 👍🏻